include/oneapi/dpl/pstl/hetero/dpcpp/execution_sycl_defs.h

```diff
@@ -311,6 +312,54 @@ using __enable_if_device_execution_policy_double_no_default =
                            __is_convertible_to_event<_Events...>,
                        _T>;
 
+template <typename _BackendTag>
+struct __hetero_tag
+{
+    using __backend_tag = _BackendTag;
+};
+
+struct __device_backend_tag
+{
+};
+
+template <typename HeteroPolicy>
+struct __select_backend_for_hetero_policy_trait;
+
+template <typename KernelName>
+struct __select_backend_for_hetero_policy_trait<execution::device_policy<KernelName>>
+{
+    using __backend_tag = __device_backend_tag;
+};
+
+#if _ONEDPL_FPGA_DEVICE
```

Comment:
Basically, it is possible to have several hetero backend simultaneously, in the same compiled unit with tag dispatching approach. It seems that we have now is not quite correct.  The preprocessor logic `#if _ONEDPL_FPGA_DEVICE` should gone...

Reply:
I believe this propose is out of scope for this PR because as you may see `execution::fpga_policy` now is defined under `#if _ONEDPL_FPGA_DEVICE` macro check.

Reply:
It is in of scope, I think.

Reply:
@rarutyun what do you think about this?

Reply:
Discussed offline.